middle_end/middle_end.ml

```diff
@@ -36,6 +36,22 @@ let middle_end ppf ~prefixname ~backend
     ~module_ident
     ~module_initializer =
   Profile.record_call "flambda" (fun () ->
+    let previous_warning_printer = !Location.warning_printer in
```

Comment:
If you keep the logic here (but I think it would be better in Location directly), you could use Misc.protect_refs (which nicely managed setting back the original value even when an exception is raised).

Reply:
Thanks for the `Misc.protect_refs` tip! Regarding the move to `Location`,
I would like to see whether other people share my concern, as I might have
been over cautious.